src/sentencepiece_trainer.cc

```diff
@@ -153,6 +153,12 @@ util::Status SentencePieceTrainer::MergeSpecsFromArgs(
       CHECK_OR_RETURN(absl::SimpleAtoi(value, &v));
       absl::SetFlag(&FLAGS_minloglevel, v);
       continue;
+    } else if(key == "encode_unicode_case") {
+      bool b;
+      std::istringstream("true") >> std::boolalpha >> b;
```

Comment:
If I understand correctly, you assume that the option is set to true if only `--encode_unicode_case` is present, regardless of the option value. What about `--encode_unicode_case=false`? Sentencepiece options seem to support this notation, but it won't work here. Maybe you can simply use `value` instead of hardcoding it to `"true"`?

Reply:
Yes, I missed that before pushing yesterday, and fixing that today was giving me some weird runtime errors. I'll test and correct this.

Reply:
Fixed.

Reply:
Great!